ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartOptionsEditor.tsx

```diff
@@ -12,12 +12,39 @@
 // limitations under the License.
 
 import { OptionsEditorProps } from '@perses-dev/plugin-system';
-import { Box } from '@mui/material';
+import { Stack, Box } from '@mui/material';
 import { TimeSeriesChartOptions } from './time-series-chart-model';
+import { TimeSeriesChartPanel, TimeSeriesChartProps } from './TimeSeriesChartPanel';
 
 export type TimeSeriesChartOptionsEditorProps = OptionsEditorProps<TimeSeriesChartOptions>;
 
 export function TimeSeriesChartOptionsEditor(props: TimeSeriesChartOptionsEditorProps) {
   const { value } = props;
-  return <Box>{JSON.stringify(value)}</Box>;
+
+  const panelPreviewProps: TimeSeriesChartProps = {
+    definition: {
+      kind: 'Panel',
+      spec: {
+        // TODO: how to exlude display (display.name is required but gets overriden)
+        display: { name: 'Temp Name' },
+        plugin: {
+          kind: 'TimeSeriesChart',
+          spec: {
+            ...value,
+          },
+        },
+      },
+    },
+    contentDimensions: { width: 500, height: 250 },
+  };
```

Comment:
At one point, I played with using the generic `Panel` in the `PanelDrawer` (at the time, it was accessible in PanelDrawer, not sure if it still is), and then it would take care of rendering the preview for every type of panel. But also not sure this approach can work fully, it was more prototype code.

Reply:
I'll look more into this too, it definitely would be a much better approach!